### PR TITLE
Ordering planner

### PR DESCRIPTION
**Pull request by b1n-ch1kn**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimisation

## [required] Description

This orderer and spline planner used to live in `sim_translators` but that pkg is redundant now, so moving it to a dedicate node which will be used not just for sim.
Added `nav_msgs` `Path` message publishing too

## [required] Documentation

Already completed with ordering method was documented

### [optional] Usability concerns or breaking changes?

Not from this PR, but identified while working - our custom markers display was removed when we moved to using rviz plugins for cone markers, but this also accidently removed the path line markers. This will be re-added, better than ever, in a new PR but for now, the nav Path message will suffice.

### CODE REVIEW

**Review by supex0fan — COMMENTED** [review]
Making it a dedicated planner is nice.

Does this subscriber need to be updated to use the QUTMSPathStamped message?
https://github.com/QUT-Motorsport/QUTMS_Driverless/blob/master/src/common/driverless_common/driverless_common/node_display.py#L45

File: src/navigation/planners/planners/node_ordered_mid_spline.py [hunk 1]
```diff
@@ -194,15 +194,39 @@ def map_callback(self, track_msg: ConeDetectionStamped):
             mid_x, mid_y = midpoint([yx[i], yy[i]], [bx[i], by[i]])
             tx.append(mid_x)
             ty.append(mid_y)
+            # for old QUTMS path curve intensity calcs
             th.append(angle([bx[i], by[i]], [yx[i], yy[i]]))  # angle of tangent at midpoint
```
Comment (supex0fan) [inline src/navigation/planners/planners/node_ordered_mid_spline.py:198]: While you are here would be good to use the /driverless_common/common.py implementation of angle().

**Review by supex0fan — CHANGES_REQUESTED** [review]
While you are here, would be good to use the driverless_common angle() method.

**Comment by b1n-ch1kn** [comment]
> Making it a dedicated planner is nice.
> 
> Does this subscriber need to be updated to use the QUTMSPathStamped message? https://github.com/QUT-Motorsport/QUTMS_Driverless/blob/master/src/common/driverless_common/driverless_common/node_display.py#L45

This is just a [rename of the import](https://github.com/QUT-Motorsport/QUTMS_Driverless/blob/4e0762acfb53e3bee9ea16bd0aeae75bcde1538c/src/navigation/planners/planners/node_ordered_mid_spline.py#L12C46-L12C46) to reduce confusion (given there were 2 "path" message classes, I specified the custom QUTMS one). 
Might actually be less confusing if I leave them hey

**Review by Damin-G — APPROVED** [review]

**Comment by supex0fan** [comment]
> > Making it a dedicated planner is nice.
> > Does this subscriber need to be updated to use the QUTMSPathStamped message? https://github.com/QUT-Motorsport/QUTMS_Driverless/blob/master/src/common/driverless_common/driverless_common/node_display.py#L45
> 
> This is just a [rename of the import](https://github.com/QUT-Motorsport/QUTMS_Driverless/blob/4e0762acfb53e3bee9ea16bd0aeae75bcde1538c/src/navigation/planners/planners/node_ordered_mid_spline.py#L12C46-L12C46) to reduce confusion (given there were 2 "path" message classes, I specified the custom QUTMS one). Might actually be less confusing if I leave them hey

I see, my bad. It might be better to add a prefix to the nav message as it is the odd one out, but I'm not fussed either way.

**Review by supex0fan — APPROVED** [review]

**Pull request merged**